### PR TITLE
[libgig] Update to 4.4.1

### PR DESCRIPTION
**Pull request by Rossmaxx**
Since the last time I added libgig to vcpkg, libgig 4.4.0 was released, where my patches were accepted upstream . Contains some stability improvements.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Rossmaxx** [comment]
If anyone is looking to this one, please hold it for now. The maintainer of libgig has decided to accept my patches and make a 4.4.1 release so might as well wait for that one instead.

File: ports/libgig/portfile.cmake [hunk 1]
```diff
@@ -1,16 +1,15 @@
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://download.linuxsampler.org/packages/libgig-4.3.0.tar.bz2"
-    FILENAME "libgig-4.3.0.tar.bz2"
-    SHA512 683c09b1d17acf69020c631452b7dfb25ac54c3701db5e97471d4e7973e9a06267667bf19bfe4eb00d2964223e8446f248d93b4cf29c062dec2588758b4dfba2
+    URLS "https://download.linuxsampler.org/packages/libgig-4.4.0.tar.bz2"
+    FILENAME "libgig-4.4.0.tar.bz2"
```
Comment (WangWeiLin-MV) [inline ports/libgig/portfile.cmake:3]:
```suggestion
    URLS "https://download.linuxsampler.org/packages/libgig-${VERSION}.tar.bz2"
    FILENAME "libgig-${VERSION}.tar.bz2"
```

**Review by WangWeiLin-MV — CHANGES_REQUESTED** [review]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Comment by Rossmaxx** [comment]
Update : 4.4.1 is released and I have bumped this pr to use that new version instead. Also applied the suggested changes by @WangWeiLin-MV

File: ports/libgig/portfile.cmake (hunk 1 above)
Comment (Rossmaxx) [reply ports/libgig/portfile.cmake]: done

**Review by WangWeiLin-MV — APPROVED** [review]

File: versions/l-/libgig.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "f08981437a0c6f00c2f885f310caf00e328d6280",
+      "version": "4.4.1",
+      "port-version": 0
+    },
+    {
+      "git-tree": "b6bf18eb1b5fb25c43059dde501b51861ff71ead",
+      "version": "4.4.0",
+      "port-version": 0
+    },
```
Comment (WangWeiLin-MV) [inline versions/l-/libgig.json:12]:
```suggestion
```
According to the checklist: "Only one version is added to each modified port's versions file."

**Review by WangWeiLin-MV — CHANGES_REQUESTED** [review]

**Review by WangWeiLin-MV — APPROVED** [review]

File: ports/libgig/portfile.cmake [hunk 3]
```diff
@@ -1,16 +1,12 @@
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://download.linuxsampler.org/packages/libgig-4.3.0.tar.bz2"
-    FILENAME "libgig-4.3.0.tar.bz2"
-    SHA512 683c09b1d17acf69020c631452b7dfb25ac54c3701db5e97471d4e7973e9a06267667bf19bfe4eb00d2964223e8446f248d93b4cf29c062dec2588758b4dfba2
+    URLS "https://download.linuxsampler.org/packages/libgig-${VERSION}.tar.bz2"
+    FILENAME "libgig-${VERSION}.bz2"
```
Comment (WangWeiLin-MV) [inline ports/libgig/portfile.cmake:3]:
```suggestion
    FILENAME "libgig-${VERSION}.tar.bz2"
```

**Review by WangWeiLin-MV — CHANGES_REQUESTED** [review]

File: ports/libgig/portfile.cmake (hunk 3 above)
Comment (Rossmaxx) [reply ports/libgig/portfile.cmake:3]: Oops mb

**Comment by WangWeiLin-MV** [comment]
@Rossmaxx The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.

**Review by WangWeiLin-MV — DISMISSED** [review]

**Comment by WangWeiLin-MV** [comment]
https://github.com/microsoft/vcpkg/blob/72d66da23b4332533c82d797bf2f82b8d6d1bf8a/ports/libgig/portfile.cmake#L18-L22
The check feature `extra-tools` should be `tools` here.

**Comment by WangWeiLin-MV** [comment]
Hi, just a suggestion, would you like to help add the optional field `license` in `vcpkg.json` of this port?
https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#license

File: ports/libgig/portfile.cmake [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
 vcpkg_extract_source_archive(
     SOURCE_PATH
     ARCHIVE "${ARCHIVE}"
-    PATCHES
-        0001-msvc-fix-ssize_t.patch
-        0002-cmake-fixes.patch
-        0003-fix-usage.patch
 )
 
 string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} dynamic LIBGIG_BUILD_SHARED)
```
Comment (WangWeiLin-MV) [inline ports/libgig/portfile.cmake:12]: https://github.com/microsoft/vcpkg/blob/72d66da23b4332533c82d797bf2f82b8d6d1bf8a/ports/libgig/portfile.cmake#L18-L22
The check feature `extra-tools` should be `tools` here.

Comment (Rossmaxx) [reply ports/libgig/portfile.cmake:12]: done

**Review by WangWeiLin-MV — APPROVED** [review]

**Comment by WangWeiLin-MV** [comment]
Port and features test pass with following triplets:
```plain
x64-windows
x64-windows-static
x86-windows
```

**Pull request merged**